skema/rest/integrated_text_reading_proxy.py

```diff
@@ -0,0 +1,209 @@
+                ),
+            }
+            response = requests.post(
+                f"{MIT_TR_ADDRESS}/integration/get_mapping", params=params, data=data
+            )
+
+            if response.status_code == 200:
+                map_data = response.text
+                map_path = os.path.join(tmpdirname, "mapping.txt")
+                with open(map_path, "w") as f:
+                    f.write(map_data)
+                merged_collection = merge_collections(
+                    a_collection=collections[0],
+                    m_collection=collections[1],
+                    map_path=Path(map_path),
+                )
+
+                # Return the merged collection here
+                return merged_collection
+
+    # Merge the collections into a attribute collection
+    attributes = list(it.chain.from_iterable(c.attributes for c in collections))
+
+    return AttributeCollection(attributes=attributes)
+
+
+# End utility code for the endpoints
+
+
+@router.post(
+    "/integrated-text-extractions",
+    summary="Posts one or more plain text documents and annotates with SKEMA and/or MIT text reading pipelines",
+)
+async def integrated_text_extractions(
+    texts: TextReadingInputDocuments,
+    annotate_skema: bool = True,
+    annotate_mit: bool = True,
+) -> TextReadingAnnotationsOutput:
+    texts = texts.texts
+    skema_extractions = [[] for t in texts]
+
+    if annotate_skema:
+        try:
+            skema_extractions = annotate_text_with_skema(texts)
+        except Exception as ex:
+            print(f"Problem annotating with Skema: {ex}")
+
+    mit_extractions = [[] for t in texts]
+
+    if annotate_mit:
+        mit_extractions = annotate_text_with_mit(texts)
+
+    results = list()
+    errors = list()
+    assert len(skema_extractions) == len(
+        mit_extractions
+    ), "Both pipeline results lists should have the same length"
+    for skema, mit in zip(skema_extractions, mit_extractions):
+        if annotate_skema and isinstance(skema, str):
+            errors.append(TextReadingError(pipeline="SKEMA", message=skema))
+
+        if annotate_mit and isinstance(mit, str):
+            errors.append(TextReadingError(pipeline="MIT", message=mit))
+
+        normalized = normalize_extractions(
+            arizona_extractions=skema, mit_extractions=mit
+        )
+        results.append(
+            TextReadingDocumentResults(
+                data=normalized if normalized.attributes else None,
+                errors=errors if errors else None,
+            )
+        )
+    return TextReadingAnnotationsOutput(outputs=results)
+
+
+@router.get(
+    "/healthcheck",
+    summary="Check health of integrated text reading service",
+    response_model=int,
+    status_code=200,
+)
+def healthcheck() -> int:
+    # SKEMA health check
+    skema_endpoint = f"{SKEMA_TR_ADDRESS}/api/skema"
+    skema_response = requests.get(skema_endpoint, timeout=10)
```

Comment:
As we're expecting, your unit test fails: https://github.com/ml4ai/skema/actions/runs/5508698590/jobs/10040336309?pr=308#step:12:329

We shouldn't expect those services to be available at test time (see https://github.com/ml4ai/skema/pull/308#issuecomment-1627485868), so we need to be more robust with error handling in this test.